src/python/pants/backend/python/rules/python_fmt.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os
+
+from pants.backend.python.rules.create_requirements_pex import (RequirementsPex,
+                                                                RequirementsPexRequest)
+from pants.backend.python.subsystems.black import Black
+from pants.backend.python.subsystems.python_setup import PythonSetup
+from pants.backend.python.subsystems.subprocess_environment import SubprocessEncodingEnvironment
+from pants.engine.fs import Digest, DirectoriesToMerge, PathGlobs, Snapshot
+from pants.engine.isolated_process import ExecuteProcessRequest, ExecuteProcessResult
+from pants.engine.legacy.structs import (PantsPluginAdaptor, PythonAppAdaptor, PythonBinaryAdaptor,
+                                         PythonTargetAdaptor, PythonTestsAdaptor)
+from pants.engine.rules import UnionRule, optionable_rule, rule
+from pants.engine.selectors import Get
+from pants.rules.core.fmt import FmtResult, FmtTarget
+from pants.util.objects import datatype
+
+
+class PythonFormatable(datatype(['target'])):
+  pass
+
+
+@rule
+def run_black(
+  target: PythonFormatable,
+  black: Black,
+  python_setup: PythonSetup,
+  subprocess_encoding_environment: SubprocessEncodingEnvironment,
+  ) -> FmtResult:
+  config_path = black.get_options().config
+  config_snapshot = yield Get(Snapshot, PathGlobs, PathGlobs(include=(config_path,)))
+
+  resolved_requirements_pex = yield Get(
+    RequirementsPex, RequirementsPexRequest(
+      output_filename="black.pex",
+      requirements=tuple(black.get_requirement_specs()),
+      interpreter_constraints=(),
```

Comment:
`PythonToolBase` by default has a flag for this, so we should probably pass it through from the `Black` instance.

Also, IIRC black has a minimum python version of 3.6, so we probably want `Black` to override `default_interpreter_constraints` to specify that minimum.

Reply:
> Also, IIRC black has a minimum python version of 3.6, so we probably want Black to override default_interpreter_constraints to specify that minimum.
By default, it picks the flag based on the file itself, so I don't think that's necessary.

Reply:
I'm not sure I follow... What would stop this pex from trying to run with py27 if that's the default python on your system?

Reply:
Sorry: misread your comment. I thought you were speaking about Black's option of specifying the version of the file that it's formatting. Addressed now.